mathics/core/atoms.py

```diff
@@ -812,6 +796,10 @@ def boxes_to_text(self, show_string_characters=False, **options) -> str:
 
         return value
 
+    # ``boxes_to_*`` are going to be removed from the ``Atom`` interface.
```

Comment:
Removing `boxes_to_*`  and replacing it with a more extensible mechanism would be awesome!

Reply:
With the last changes, we can remove all these methods, except for the ones in String. Once this get merged, that is the next step